runtime/jcl/common/reflecthelp.c

```diff
@@ -833,9 +833,17 @@ createField(struct J9VMThread *vmThread, jfieldID fieldID)
 #if JAVA_SPEC_VERSION >= 15
 	/* trust that static final fields and final record or hidden class fields will not be modified. */
 	if (J9_ARE_ALL_BITS_SET(j9FieldID->field->modifiers, J9AccFinal)) {
+#if JAVA_SPEC_VERSION >= 16
```

Comment:
This would be better achieved by putting the Record class in the VM constant pool, rather than doing a name check.